### PR TITLE
[MI-2904]:Fixed issue #512 and #513 on github plugin

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
https://github.com/mattermost/mattermost-plugin-github/issues/512
https://github.com/mattermost/mattermost-plugin-github/issues/513

### CODE REVIEW

File: server/plugin/command.go [hunk 1]
```diff
@@ -318,22 +338,54 @@ func (p *Plugin) handleSubscribesAdd(_ *plugin.Context, args *model.CommandArgs,
 
 	ctx := context.Background()
 	githubClient := p.githubConnectUser(ctx, userInfo)
+	user, appErr := p.API.GetUser(args.UserId)
+	if appErr != nil {
+		return appErr.Error()
+	}
+	owner, repo := parseOwnerAndRepo(parameters[0], baseURL)
```
Comment (ayusht2810) [inline server/plugin/command.go]:
```suggestion
	}
	
	owner, repo := parseOwnerAndRepo(parameters[0], baseURL)
```

File: server/plugin/command.go [hunk 2]
```diff
@@ -318,22 +338,54 @@ func (p *Plugin) handleSubscribesAdd(_ *plugin.Context, args *model.CommandArgs,
...
+
+		post := &model.Post{
+			ChannelId: args.ChannelId,
+			UserId:    p.BotUserID,
+			Message:   subscriptionSuccess,
+		}
 
-		return fmt.Sprintf("Successfully subscribed to organization %s.", owner)
+		if _, appErr = p.API.CreatePost(post); appErr != nil {
+			p.API.LogWarn("error while creating post", "post", post, "error", appErr.Error())
```
Comment (ayusht2810) [inline server/plugin/command.go]:
```suggestion
			p.API.LogWarn("Error while creating post", "Post", post, "Error", appErr.Error())
```

File: server/plugin/command.go [hunk 3]
```diff
@@ -342,22 +394,98 @@ func (p *Plugin) handleSubscribesAdd(_ *plugin.Context, args *model.CommandArgs,
...
 
-	return msg
+	post := &model.Post{
+		ChannelId: args.ChannelId,
+		UserId:    p.BotUserID,
+		Message:   msg,
+	}
+
+	if _, appErr := p.API.CreatePost(post); appErr != nil {
+		p.API.LogWarn("error while creating post", "post", post, "error", appErr.Error())
```
Comment (ayusht2810) [inline server/plugin/command.go]: same

File: server/plugin/command.go [hunk 4]
```diff
@@ -342,22 +394,98 @@ func (p *Plugin) handleSubscribesAdd(_ *plugin.Context, args *model.CommandArgs,
...
+}
 func (p *Plugin) handleUnsubscribe(_ *plugin.Context, args *model.CommandArgs, parameters []string, _ *GitHubUserInfo) string {
 	if len(parameters) == 0 {
 		return "Please specify a repository."
 	}
 
 	repo := parameters[0]
+	config := p.getConfiguration()
+	owner, repo := parseOwnerAndRepo(repo, config.getBaseURL())
+	if owner == "" && repo == "" {
```
Comment (ayusht2810) [inline server/plugin/command.go:442]: should this condition be `||` or `&&`?

File: server/plugin/command.go [hunk 5]
```diff
@@ -342,22 +394,98 @@ func (p *Plugin) handleSubscribesAdd(_ *plugin.Context, args *model.CommandArgs,
...
+	if err := p.Unsubscribe(args.ChannelId, repo, owner); err != nil {
 		p.API.LogWarn("Failed to unsubscribe", "repo", repo, "error", err.Error())
 		return "Encountered an error trying to unsubscribe. Please try again."
 	}
 
-	return fmt.Sprintf("Successfully unsubscribed from %s.", repo)
+	baseURL := config.getBaseURL()
+	user, appErr := p.API.GetUser(args.UserId)
+	if appErr != nil {
+		p.API.LogWarn("error while fetching user details", "error", appErr.Error())
```
Comment (ayusht2810) [inline server/plugin/command.go]: same. update everywhere, if changing

File: server/plugin/command.go [hunk 6]
```diff
@@ -342,22 +394,98 @@ func (p *Plugin) handleSubscribesAdd(_ *plugin.Context, args *model.CommandArgs,
...
+		unsubscriptionMessage = fmt.Sprintf("@%v unsubscribed this channel from [%s](%s)", user.Username, owner, orgLink)
+		post := &model.Post{
+			ChannelId: args.ChannelId,
+			UserId:    p.BotUserID,
+			Message:   unsubscriptionMessage,
+		}
+
+		if _, appErr := p.API.CreatePost(post); appErr != nil {
+			p.API.LogWarn("error while creating post", "post", post, "error", appErr.Error())
+			return fmt.Sprintf("%s Though there was an error creating the public post: %s", unsubscriptionMessage, appErr.Error())
```
Comment (ayusht2810) [inline server/plugin/command.go]: is this message correct?

File: server/plugin/command.go (hunk 4 above)
Comment (Kshitij-Katiyar) [reply server/plugin/command.go:442]: It is correct

**Review by ayusht2810 — APPROVED** [review]

File: server/plugin/command.go [hunk 7]
```diff
@@ -38,6 +38,27 @@ var validFeatures = map[string]bool{
 	featureStars:         true,
 }
 
+const (
+	subCommandListDeleteAll = "delete-all"
```
Comment (manojmalik20) [inline server/plugin/command.go]:
```suggestion
	subCommandDeleteAll = "delete-all"
```

File: server/plugin/command.go [hunk 8]
```diff
@@ -318,22 +338,55 @@ func (p *Plugin) handleSubscribesAdd(_ *plugin.Context, args *model.CommandArgs,
...
+			ChannelId: args.ChannelId,
+			UserId:    p.BotUserID,
+			Message:   subscriptionSuccess,
+		}
 
-		return fmt.Sprintf("Successfully subscribed to organization %s.", owner)
+		if _, appErr = p.API.CreatePost(post); appErr != nil {
+			p.API.LogWarn("Error while creating post", "Post", post, "Error", appErr.Error())
+			return fmt.Sprintf("%s error creating the public post: %s", subscriptionSuccess, appErr.Error())
+		}
```
Comment (manojmalik20) [inline server/plugin/command.go:377]:
```suggestion
		}
		
```

File: server/plugin/command.go [hunk 9]
```diff
@@ -342,22 +395,98 @@ func (p *Plugin) handleSubscribesAdd(_ *plugin.Context, args *model.CommandArgs,
...
+
+	if _, appErr := p.API.CreatePost(post); appErr != nil {
+		p.API.LogWarn("Error while creating post", "Post", post, "Error", appErr.Error())
+		return fmt.Sprintf("%s error creating the public post: %s", msg, appErr.Error())
+	}
+
+	return ""
 }
 
+func (p *Plugin) getSubscribedFeatures(channelID string, owner, repo string) (Features, error) {
```
Comment (manojmalik20) [inline server/plugin/command.go]:
```suggestion
func (p *Plugin) getSubscribedFeatures(channelID, owner, repo string) (Features, error) {
```

File: server/plugin/command.go [hunk 10]
```diff
@@ -342,22 +395,98 @@ func (p *Plugin) handleSubscribesAdd(_ *plugin.Context, args *model.CommandArgs,
...
+		label := repo
+		if owner != "" {
+			label = owner + "/" + repo
+		}
+
+		if sub.Repository == label {
+			previousFeatures = sub.Features
+			return previousFeatures, nil
+		}
+	}
```
Comment (manojmalik20) [inline server/plugin/command.go:430]:
```suggestion
	}
	
```

File: server/plugin/command.go [hunk 11]
```diff
@@ -342,22 +395,98 @@ func (p *Plugin) handleSubscribesAdd(_ *plugin.Context, args *model.CommandArgs,
...
 
-	return fmt.Sprintf("Successfully unsubscribed from %s.", repo)
+	baseURL := config.getBaseURL()
+	user, appErr := p.API.GetUser(args.UserId)
+	if appErr != nil {
+		p.API.LogWarn("Error while fetching user details", "Error", appErr.Error())
+		return fmt.Sprintf("error while fetching user details: %s", appErr.Error())
+	}
+
+	unsubscriptionMessage := ""
```
Comment (manojmalik20) [inline server/plugin/command.go]:
```suggestion
	unsubscribeMessage := ""
```

File: server/plugin/subscriptions.go [hunk 12]
```diff
@@ -324,17 +324,7 @@ func (p *Plugin) GetSubscribedChannelsForRepository(repo *github.Repository) []*
...
-
-	owner, repo := parseOwnerAndRepo(repo, config.getBaseURL())
-	if owner == "" && repo == "" {
-		return errors.New("invalid repository")
-	}
-
-	owner = strings.ToLower(owner)
-	repo = strings.ToLower(repo)
-
+func (p *Plugin) Unsubscribe(channelID string, repo, owner string) error {
```
Comment (manojmalik20) [inline server/plugin/subscriptions.go]:
```suggestion
func (p *Plugin) Unsubscribe(channelID, repo, owner string) error {
```

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**